web/html/src/manager/virtualization/guests/console/guests-console.js

```diff
@@ -219,6 +223,21 @@ class GuestsConsole extends React.Component<Props, State> {
     );
   };
 
+  updateGuestData = () => {
+    Network.get(`/rhn/manager/api/systems/details/virtualization/guests/${this.state.hostId}/guest/${this.props.guestUuid}`,
+      'application/json').promise
+      .then((response) => {
+        this.setState({
+          graphicsType: response.graphics?.type,
+          guestName: response.name,
+        }, this.refreshToken);
+      }, (xhr) => {
+        // We may have hit the time where the machine is not yet in the DB, try again
+        setTimeout(() => {}, 500);
```

Comment:
In Javascript, this will not pause the execution and the next line `this.updateGuestData()` will be called immediately. If you want to call the function after half a second, you could do the following instead:

```js
setTimeout(() => {
  this.updateGuestData();
}, 500);
```

However, this is not really safe as this may lead to a situation where the component is unmounted and the timeout resolves after the fact. To avoid the problem, you can keep track of the timeout id returned by `setTimeout` and use `clearTimeout(timeoutId)` when the component is unmounting to ensure there will be no race condition.  

```js
class Foo {
  barTimeoutId = null;

  componentWillUnmount() {
    if (this.barTimeoutId) {
      clearTimeout(this.barTimeoutId);
    }
  }

  updateBar = () => {
    Network.get(...)
      .then(...)
      .catch(xhr => {
        if (xhr.status === ???) {
          if (this.barTimeoutId) {
            clearTimeout(this.barTimeoutId);
          }
          this.barTimeoutId = setTimeout(..., 500);
        }
      });
  }
}
```

Reply:
ACK